source/com.microsoft.tfs.core/src/com/microsoft/tfs/core/telemetry/TfsTelemetryHelper.java

```diff
@@ -44,8 +50,21 @@
             TfsTelemetryInstrumentationInfo.isDeveloperMode());
     }
 
+    public synchronized static void checkDisableTelemetryProperty() {
+        // Unfortunately, we can't push the disabled logic onto the
+        // TelemetryClient itself because the initialization code for
+        // that object simply takes too much time. So, we have to stop
+        // the sending of telemetry here in the helper class.
+
+        // If the system property exists or the env var is true, then we disable
+        // telemetry.
+        sendingTelemetryDisabled = System.getProperty(DISABLE_TELEMETRY_PROPERTY_NAME) != null
```

Comment:
Correct. I am using it as a flag. If this is not ideal, I can remove the System Property. I ended up using the env var from IntelliJ - it was just easier. I kept the System property because I thought we might end up going that route if we need to later.